lib/chef/data_bag.rb

```diff
@@ -42,6 +42,9 @@ def self.validate_name!(name)
       if RESERVED_NAMES.match?(name)
         raise Exceptions::InvalidDataBagName, "DataBags may not have a name matching #{RESERVED_NAMES.inspect}, you gave #{name.inspect}"
       end
+      if name.include? "."
+        Chef::Log.warn(" Databags should not contain ' . ' period in name, you gave: #{name.inspect}")
```

Comment:
this should probably be a deprecation so it shows at the end and gets de-duped and eventually the support for it gets removed.